### ISSUE TITLE
Setting up a host for a directive with providers

### ISSUE THREAD

**Issue by Dyljyn** [NetanelBasal/spectator#13]
### Problem

I have a use case where a directive makes use of a service. 
However, I can not figure out how to set this up with `createHostComponentFactory`.

> I've used `AppUnlessDirective` from the [Angular docs] as an example.

```js
const createHost = createHostComponentFactory({
  component: AppUnlessDirective,
  providers: [AppUnlessService]
});
```

Giving the following error (because it's a directive and not a component?)

```
TypeError: Cannot read property 'componentInstance' of null
```

### Example code

https://stackblitz.com/edit/spectator-directive-providers?file=app%2Fapp-unless.directive.spec.ts

[Angular docs]: https://angular.io/guide/structural-directives#write-a-structural-directive

**Comment by NetanelBasal** [comment]
I see where is the problem, let me investigate.

**Comment by NetanelBasal** [comment]
It's not related to the service, it's related to the structural directive.

**Comment by NetanelBasal** [comment]
Fixed in 1.4.4

https://stackblitz.com/edit/spectator-structural-directive-y16jdw?file=app/app-unless.directive.spec.ts

Thanks!
